### PR TITLE
[test]: user sign up endpoint

### PR DESCRIPTION
**Pull request by yovanny-wolox**
## Summary

- added tests for endpoint signup POST /users
- mock and factory data were generated.
- removed mocha package
- updated jest, supertest and nodemon packages
- added migrations-test script

## Known Issues

none

## Trello Card

https://trello.com/c/X0AG584F/22-tests-de-sign-up

## Screenshot
tests

### CODE REVIEW

File: test/controllers/users.test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    test('failed with an email already exists', () =>
+      userFactory
+        .create()
+        .then(() =>
+          request(app)
+            .post(USERS_PATH)
+            .set(DEFAULT_HEADER)
+            .send(userTestInfo)
+        )
+        .then(res => expect(res.statusCode).toBe(CONFLICT_CODE)));
```
Comment (jersonlopez) [inline test/controllers/users.test.js]: why aren't you using `.catch` here instead of `.then`?

Comment (jersonlopez) [reply test/controllers/users.test.js]: this comment applies to all failed cases tests

Comment (yovanny-wolox) [reply test/controllers/users.test.js]: I generally don't put catch in tests files. Thanks by feedback!

Comment (yovanny-wolox) [reply test/controllers/users.test.js]: Done!

**Review by jersonlopez — APPROVED** [review]
Good Job @yovanny-wolox 💯

**Pull request merged**